create_commits.py

```diff
@@ -0,0 +1,347 @@
+    """
+
+    body: str
+    headline: str
+
+
+def get_file_changes_from_local_commit_hash(commit_hash: str) -> FileChanges:
+    """
+    Create a file changes object for a specific commit.
+
+    Args:
+        commit_hash (str): The hash of the commit.
+
+    Returns:
+        dict: A dictionary representing the FileChanges object.
+    """
+
+    # setting config variable advice.detachedHead to false to avoid verbose messages in the logs
+    subprocess.run(
+        ["git", "config", "advice.detachedHead", "false"],
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+
+    logging.debug("")
+    logging.debug("now working on commit hash %s", commit_hash)
+    # Get a list of files changed in a specific commit.
+    result = subprocess.run(
+        ["git", "diff", "--name-status", f"{commit_hash}~1", commit_hash],
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    files_changed_by_commit = result.stdout.splitlines()
+
+    logging.debug("files_changed_by_commit:\n %s", files_changed_by_commit)
+    file_changes: FileChanges = {
+        "additions": [],
+        "deletions": [],
+    }
+
+    # check out that commit in detached head state, so that we can pull file contents accurately
+    subprocess.run(["git", "checkout", commit_hash], check=True)
+    logging.debug("output of ls after checking out commit hash:")
+    subprocess.run(["ls"], check=True)
+
+    for line in files_changed_by_commit:
+        status, *filenames = line.split("\t")
+        logging.debug("status: %s", status)
+        logging.debug("filenames: %s", filenames)
+        if status in ["A", "M"]:
+            logging.debug("Added or modified file detected")
+            with open(filenames[-1], "rb") as f:
+                contents = base64.b64encode(f.read()).decode("utf-8")
+            file_changes["additions"].append(
+                FileAddition(
+                    path=filenames[-1],
+                    contents=contents,
+                )
+            )
+        elif "R" in status or status == "R":
+            logging.debug("Renamed file detected")
+            old_name = filenames[0]
+            new_name = filenames[1]
+            logging.debug("old_name: %s", old_name)
+            logging.debug("new_name: %s", new_name)
+
+            file_changes["deletions"].append({"path": old_name})
+            with open(new_name, "rb") as f:
+                contents = base64.b64encode(f.read()).decode("utf-8")
+            file_changes["additions"].append(
+                FileAddition(
+                    path=new_name,
+                    contents=contents,
+                )
+            )
+
+        elif status in ["D"]:
+            logging.debug("Deleted file detected")
+            file_changes["deletions"].append({"path": filenames[0]})
+
+    # go back to the previous ref
+    subprocess.run(["git", "checkout", "-"], check=True, capture_output=True, text=True)
```

Comment:
i usually recommend avoiding `capture_output` unless you specifically want to capture `stderr`, because otherwise it makes things harder to debug.
```suggestion
    subprocess.run(["git", "checkout", "-"], check=True, stdout=subprocess.PIPE, text=True)
```

Reply:
ooh thank you! this was copilot - I haven't used `subprocess.run` a ton yet, so this is some veteran know-how that I appreciate a lot heh